torch_xla/experimental/xla_sharding.py

```diff
@@ -104,9 +105,15 @@ def get_op_sharding(self,
     replicate_dims = {i for i, d in enumerate(partition_spec) if d is None}
     group_assignment, replication_groups = _get_group_assignment(
         sharding_type, tile_assignment, len(partition_spec), replicate_dims)
-    return torch_xla._XLAC.OpSharding(tile_assignment.tolist(),
-                                      group_assignment, replication_groups,
-                                      int(sharding_type))
+
+    # If flatten_opsharding = True, return the flattened version of OpSharding
+    if flatten_opsharding:
```

Comment:
What is flattening here? Just returning as a tuple? If so, maybe call it `as_tuple`. I am still hesitant to override the return type here... you can try this using the accessor methods instead, since you only need the `sharding_type` value in your work. Let me create a follow-up PR and add you, let's land this for now. Thanks @wonjoolee95